server/plugins/client/bundle-handler-factory.js

```diff
@@ -0,0 +1,54 @@
+module.exports.createHandler = createHandler
+
+var fs = require('fs')
+
+var log = require('npmlog')
+var bundleClient = require('./bundle')
+
+function createHandler (hoodieClientPath, bundleTargetPath, bundleConfig) {
+  var bundlePromise
+
+  return function (request, reply) {
+    if (!bundlePromise) {
+      bundlePromise = new Promise(function (resolve, reject) {
+        log.silly('client', 'bundling')
+        bundleClient(hoodieClientPath, bundleTargetPath, bundleConfig, function (error, bundleBuffer, hasUpdate) {
+          if (error) {
+            return reject(error)
+          }
+
+          writeClientBundle(hasUpdate, bundleConfig.inMemory, bundleTargetPath, bundleBuffer)
+
+          resolve(bundleBuffer)
+        })
+      })
+    }
+
+    bundlePromise.then(function (buffer) {
+      reply(buffer).bytes(buffer.length).type('application/javascript')
+    }).catch(reply)
+  }
+}
```

Comment:
I usually prefer factories for the reason as the word 'factory' itself already makes it pretty clear what they are doing. :)

What weighs even stronger is consistency with the rest of the code. In this particular case we have to consider one thing though: `bundlePromise` is now kind of "global" within the module. I don't assume it is a use case right now, but if you would ` require` the createHandler twice from different clients, the second handler would already find the `bundlePromise` not being undefined and therefore never call `bundleClient` (with a maybe different configuration than the first).